be/src/storage/vectorized/delta_writer.cpp

```diff
@@ -199,24 +192,41 @@ Status DeltaWriter::write(const Chunk& chunk, const uint32_t* indexes, uint32_t
     case kAborted:
     case kClosed:
         return Status::InternalError(
-                fmt::format("Fail to write delta. tablet_id: {}, state: {}", _opt.tablet_id, _state_name(state)));
+                fmt::format("Fail to prepare. tablet_id: {}, state: {}", _opt.tablet_id, _state_name(state)));
     case kWriting:
-        bool full = _mem_table->insert(chunk, indexes, from, size);
-        if (_mem_tracker->limit_exceeded()) {
-            VLOG(2) << "Flushing memory table due to memory limit exceeded";
-            st = _flush_memtable();
-            _reset_mem_table();
-        } else if (_mem_tracker->parent() && _mem_tracker->parent()->limit_exceeded()) {
-            VLOG(2) << "Flushing memory table due to parent memory limit exceeded";
-            st = _flush_memtable();
-            _reset_mem_table();
-        } else if (full) {
-            st = _flush_memtable_async();
-            _reset_mem_table();
-        }
+        return Status::OK();
+    case kPrepared: {
+        std::lock_guard push_lock(_tablet->get_push_lock());
+        st = _storage_engine->txn_manager()->prepare_txn(_opt.partition_id, _tablet, _opt.txn_id, _opt.load_id);
```

Comment:
It's also hold migration_lock when prepare_txn().

Reply:
> It's also hold migration_lock when prepare_txn().

done, changed